mathics/builtin/logic.py

```diff
@@ -116,9 +116,139 @@ class Not(PrefixOperator):
+        if result0 == Symbol('False'):
+            return Symbol('True')
+        elif result0.is_true():
+            return y.evaluate(evaluation)
+        else:
+            return Expression('Implies', result0, y.evaluate(evaluation))
+
+
+class Equivalent(BinaryOperator):
+    """
+     <dl>
+     <dt>'Equivalent[$expr1$, $expr2$, ...]'
+     <dt>'$expr1$ \u21D4 $expr2$ \u21D4 ...'
+         <dd>is equivalent to
+         ($expr1$ && $expr2$ && ...) || (!$expr1$ && !$expr2$ && ...)
+     </dl>
+
+     >> Equivalent[True, True, False]
+      = False
+
+     If all expressions do not evaluate to 'True' or 'False', 'Equivalent'
+     returns a result in symbolic form:
+     >> Equivalent[a, b, c]
+      = a \u21D4 b \u21D4 c
+      Otherwise, 'Equivalent' returns a result in DNF
+      >> Equivalent[a, b, True, c]
+       = a && b && c
+     """
+
+    operator = '\u21D4'
+    precedence = 205
+    attributes = ('Flat', 'HoldAll', 'OneIdentity')
+
+    def apply(self, args, evaluation):
+        'Equivalent[args___]'
+
+        args = args.get_sequence()
+        flag = False
+        for arg in args:
+            result = arg.evaluate(evaluation)
+            if result == Symbol('False') or result.is_true():
+                flag = not flag
+                break
+        if flag:
+            return Expression('Or', Expression('And', *args), Expression('And', *[Expression('Not', arg) for arg in args])).evaluate(evaluation)
+        else:
+            return Expression('Equivalent', *args)
+
+
+class Xor(BinaryOperator):
+    """
+    <dl>
+    <dt>'Xor[$expr1$, $expr2$, ...]'
+    <dt>'$expr1$ \u22BB $expr2$ \u22BB ...'
+        <dd>evaluates each expression in turn, returning 'True'
+        as soon as not all expressions evaluate to the same value. If all
+        expressions evaluate to the same value, 'Xor' returns 'False'.
+    </dl>
+
+    >> Xor[False, True]
+     = True
+
+    If an expression does not evaluate to 'True' or 'False', 'Xor'
+    returns a result in symbolic form:
+    >> Xor[a, False, b]
+     = a \u22BB b
+    """
+
+    operator = '\u22BB'
+    precedence = 215
+    attributes = ('Flat', 'HoldAll', 'OneIdentity')
+
+    def apply(self, args, evaluation):
+        'Xor[args___]'
+
+        args = args.get_sequence()
+        leaves = []
+        flag = True
+        for arg in args:
+            result = arg.evaluate(evaluation)
+            if result.is_true():
+                flag = not flag
+            elif result != Symbol('False'):
+                leaves.append(result)
+        if leaves and flag:
+            if len(leaves) == 1:
+                return leaves[0]
+            else:
+                return Expression('Xor', *leaves)
+        elif leaves and not flag:
+            if len(leaves) == 1:
+                return Expression('Not', leaves[0])
+            else:
+                return Expression('Not', Expression('Xor', *leaves))
+        else:
+            return Symbol(repr(not flag))
+
+
```

Comment:
Looks correct but I'd like to see some more tests. At least: `Xor[a]`, `Xor[]`, `Xor[False]`, `Xor[True]`, `Xor[a, b]`.